block-producer/src/txqueue/mod.rs

```diff
@@ -160,16 +162,20 @@ where
 
         let tx_groups = txs.chunks(self.options.batch_size).map(|txs| txs.to_vec());
 
-        for mut txs in tx_groups {
+        for txs in tx_groups {
             let ready_queue = self.ready_queue.clone();
             let batch_builder = self.batch_builder.clone();
 
             tokio::spawn(async move {
-                match batch_builder.build_batch(txs.clone()).await {
+                match batch_builder.build_batch(txs).await {
                     Ok(_) => {
                         // batch was successfully built, do nothing
                     },
-                    Err(_) => {
+                    Err(BuildBatchError::TooManyNotesCreated(_, mut txs)) => {
+                        // batch building failed, add txs back at the end of the queue
+                        ready_queue.write().await.append(&mut txs);
+                    },
+                    Err(BuildBatchError::NotesSmtError(_, mut txs)) => {
                         // batch building failed, add txs back at the end of the queue
                         ready_queue.write().await.append(&mut txs);
                     },
```

Comment:
Would it make sense to implement something like `BuildBatchError::into_transactions()` to get the list of transactions regardless of the variant? This way, we won't have to modify this code as more variants are added `BuildBatchError` in the future.

Reply:
done